plugins/outputs/application_insights/transmitter.go

```diff
@@ -1,13 +1,21 @@
 package application_insights
 
-import "github.com/Microsoft/ApplicationInsights-Go/appinsights"
+import (
+	"github.com/Microsoft/ApplicationInsights-Go/appinsights"
+)
 
 type Transmitter struct {
 	client appinsights.TelemetryClient
 }
 
-func NewTransmitter(ikey string) *Transmitter {
-	return &Transmitter{client: appinsights.NewTelemetryClient(ikey)}
+func NewTransmitter(ikey string, endpointURL string) *Transmitter {
+	if len(endpointURL) == 0 {
```

Comment:
You could simplify this function to:
```
func NewTransmitter(ikey string, endpointURL string) *Transmitter {
	telemetryConfig := appinsights.NewTelemetryConfiguration(ikey)
	telemetryConfig.EndpointUrl = endpointURL
	return &Transmitter{client: appinsights.NewTelemetryClientFromConfig(telemetryConfig)}		 
}
```
since telemetryConfig.EndpointUrl is likely already an empty string.

Reply:
`telemetryConfig.EndpointUrl` by default = `https://dc.services.visualstudio.com/v2/track` and if we set it to empty we get error